lib/src/core/guild/auto_moderation.dart

```diff
@@ -0,0 +1,305 @@
+import 'package:nyxx/nyxx.dart';
+import 'package:nyxx/src/core/snowflake.dart';
+import 'package:nyxx/src/core/snowflake_entity.dart';
+import 'package:nyxx/src/internal/cache/cacheable.dart';
+import 'package:nyxx/src/internal/exceptions/unknown_enum_value.dart';
+import 'package:nyxx/src/typedefs.dart';
+
+abstract class IAutoModerationRule implements SnowflakeEntity {
+  /// The guild's id this rule is applied to.
+  Cacheable<Snowflake, IGuild> get guild;
+
+  /// The name of this rule.
+  String get name;
+
+  /// The user which first created this rule.
+  Cacheable<Snowflake, IMember> get creator;
+
+  /// The rule event type.
+  EventTypes get eventType;
+
+  /// The rule trigger type.
+  TriggerTypes get triggerType;
+
+  /// The trigger metadata.
+  ITriggerMetadata get triggerMetadata;
+
+  /// The actions which will execute when the rule is triggered.
+  List<IActionStructure> get actions;
+
+  /// Whether this rule is enabled.
+  bool get enabled;
+
+  /// The role ids that should not be affected by the rule (Maximum of 20).
+  Iterable<RoleCacheable> get ignoredRoles;
+
+  /// The channel ids that should not be affected by the rule (Maximum of 50).
+  Iterable<ChannelCacheable<ITextGuildChannel>> get ignoredChannels;
+}
+
+enum EventTypes {
+  /// When a member sends or edits a message in a guild.
+  messageSend(1);
+
+  final int value;
+  const EventTypes(this.value);
+
+  static EventTypes _fromValue(int value) {
+    switch (value) {
+      case 1:
+        return EventTypes.messageSend;
+      default:
+        throw UnknownEnumValue(value.toString());
+    }
```

Comment:
This can be rewritten to not require changing when new event types are added:
```suggestion
    return values.firstWhere((event) => event.value == value, orElse: () => throw UnknownEnumValue(value.toString()));
```

Reply:
Nice catch! Didn't thought at this